docs/Building.md

```diff
@@ -0,0 +1,230 @@
+
+This is a more detailed guide on how to build ROS 2 packages written in Rust that expands on the minimal steps in the README.
+
+In this guide, the Foxy distribution of ROS 2 is used, but newer distributions can be used by simply replacing 'foxy' with the distribution name everywhere.
+
+
+## Environment setup
+
+Building `rclrs` requires a standard [ROS 2 installation](https://docs.ros.org/en/foxy/Installation.html), and a few extensions.
+These extensions are: `colcon-cargo`, `colcon-ros-cargo`, `cargo-ament-build`. The former two are `colcon` plugins, and the latter is a `cargo` plugin.
+
+The `libclang` library is also required for automatically generating FFI bindings with `bindgen`. See the [`bindgen` docs](https://rust-lang.github.io/rust-bindgen/requirements.html) on how to install it. As a side note, on Ubuntu the `clang` package is not required, only the `libclang-dev` package.
+
+The `python3-vcstool` package is used in [importing auxiliary repositories](#importing-repositories). It can also be installed through `pip` instead of `apt`.
+
+You can use the [premade Docker image](#using-the-docker-image) that contains all these dependencies.
+
+Otherwise, to sum up, here is how you would install the dependencies on Ubuntu:
+
+<!--- These steps should be kept in sync with README.md --->
+```shell
+# Install Rust, e.g. as described in https://rustup.rs/
+# Install ROS 2 as described in https://docs.ros.org/en/foxy/Installation.html
+# Assuming you installed the minimal version of ROS 2, you need these additional packages:
+sudo apt install -y git libclang-dev python3-pip python3-vcstool # libclang-dev is required by bindgen
+# Install these plugins for cargo and colcon:
+cargo install cargo-ament-build
+pip install git+https://github.com/colcon/colcon-cargo.git git+https://github.com/colcon/colcon-ros-cargo.git
+```
+
+
+### Choosing a workspace directory and cloning `ros2_rust`
+
+A "workspace directory", or just "workspace", is simply a directory of your choosing that contains your `ros2_rust` checkout and potentially other ROS 2 packages. It will also usually be your working directory for building. There is only one limitation: It must not contain the ROS 2 installation, so it can't be `/`, for instance. Note that this has nothing to do with a [`cargo` workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html).
+
+If you don't have a workspace directory already, simply create an empty directory in a convenient location.
+
+Next, clone `ros2_rust` into it. The rest of this article will assume that you have cloned it into a subdirectory named `src` like this:
+
+```shell
+# Make sure to run this in the workspace directory
+mkdir src
+git clone https://github.com/ros2-rust/ros2_rust.git src/ros2_rust
+```
+
+*Note: Once `rclrs` is published on crates.io, it's not technically needed anymore to clone the `ros2_rust` repo, and this section will be modified to reflect that.*
+
+### Using the Docker image
```

Comment:
```suggestion
### ALTERNATIVE: Using the Docker image
```
This suggestion may not be needed.

Reply:
You're right that it wasn't clear that this is not always required. I restructured it so that the "manual installation" and "docker" sections are next to each other, and marked as alternatives.